app/gui/view/welcome-screen/src/side_menu.rs

```diff
@@ -4,41 +4,67 @@
 
 use ensogl::prelude::*;
 
-use crate::ClickClosure;
+use crate::ClickableElement;
 
+use enso_frp as frp;
 use ensogl::system::web;
 use ensogl::system::web::NodeInserter;
 use web_sys::Element;
 
 
 
 // ================
-// === SideMenu ===
+// === Model ===
 // ================
 
-/// Side menu for Welcome Screen. Contains a list of available projects and a "new project" button.
 #[derive(Clone, CloneRef, Debug)]
-pub struct SideMenu {
+pub struct Model {
     logger:             Logger,
     pub root_dom:       Element,
-    new_project_button: Element,
-    projects_list:      Element,
-    closures:           Rc<RefCell<Vec<ClickClosure>>>,
+    new_project_button: ClickableElement,
+    projects_list_dom:  Element,
+    projects:           Rc<RefCell<Vec<ClickableElement>>>,
 }
 
-impl SideMenu {
+impl Model {
     /// Constructor.
-    pub fn new(logger: &Logger) -> Self {
-        let logger = Logger::new_sub(logger, "SideMenu");
+    pub fn new(logger: Logger) -> Self {
         let root_dom = web::create_element("aside");
         root_dom.set_class_name(crate::css_class::SIDE_MENU);
         let header = Self::create_header("Your projects");
         root_dom.append_or_warn(&header, &logger);
-        let projects_list = Self::create_projects_list();
-        root_dom.append_or_warn(&projects_list, &logger);
-        let new_project_button = Self::create_new_project_button(&logger, &projects_list);
+        let projects_list_dom = Self::create_projects_list();
+        root_dom.append_or_warn(&projects_list_dom, &logger);
+        let new_project_button = Self::create_new_project_button(&logger, &projects_list_dom);
+        let projects = default();
 
-        Self { logger, root_dom, projects_list, new_project_button, closures: default() }
+        Self { logger, root_dom, projects_list_dom, projects, new_project_button }
+    }
+
+    pub fn clear_list(&self) {
+        for entry in self.projects.borrow_mut().iter() {
+            entry.element.remove();
+        }
+        self.projects.borrow_mut().clear();
+    }
+
+    pub fn add_entry(&self, name: &str, open_project: &frp::Any<String>) {
+        let entry = Self::create_project_list_entry(name, &self.logger);
+        let network = &entry.network;
+        frp::extend! { network
+            open_project <+ entry.click.constant(name.to_owned());
+        }
```

Comment:
I don't like that attaching click events to `open_project` is implemented here, inside Model. It would be much nicer to implement it in `SideMenu::new`, in the FRP definition, but I wasn't able to find a way to do so. Besides, GraphEditor uses a similar approach.

Reply:
Lets leave it for now. This is a very valid point and Ive already seen a few similar places - should be covered in FRP 3.0